[Bug]: Study groups fails when date column includes "Nächste"

### What happened?

Error:
```
I/flutter ( 5278): 2025-01-27 00:10:34 [ERROR] Lanis: FormatException: Invalid date format
I/flutter ( 5278): 2025
I/flutter ( 5278):                                                                       ...
I/flutter ( 5278): Lanis
I/flutter ( 5278): #0      DateTime.parse (dart:core/date_time.dart:351:7)
I/flutter ( 5278): #1      StudyGroupsStudentParser.getHome.<anonymous closure> (package:sph_plan/applets/study_groups/student/parser.dart:57:34)
I/flutter ( 5278): #2      MappedListIterable.elementAt (dart:_internal/iterable.dart:435:31)
I/flutter ( 5278): #3      ListIterator.moveNext (dart:_internal/iterable.dart:364:26)
I/flutter ( 5278): #4      new _GrowableList._ofEfficientLengthIterable (dart:core-patch/growable_array.dart:189:27)
I/flutter ( 5278): #5      new _GrowableList.of (dart:core-patch/growable_array.dart:150:28)
I/flutter ( 5278): #6      new List.of (dart:core-patch/array_patch.dart:39:18)
I/flutter ( 5278): #7      ListIterable.toList (dart:_internal/iterable.dart:224:7)
I/flutter ( 5278): #8      StudyGroupsStudentParser.getHome (package:sph_plan/applets/study_groups/student/parser.dart:63:14)
I/flutter ( 5278): <asynchronous suspension>
I/flutter ( 5278): #9      AppletParser._getHome (package:sph_plan/core/applet_parser.dart:104:23)
I/flutter ( 5278): <asynchronous suspension>
I/flutter ( 5278): #10     AppletParser.fetchData.<anonymous closure> (package:sph_plan/core/applet_parser.dart:85:23)
```

### Screenshots

![Image](https://github.com/user-attachments/assets/e9a3247e-4f25-451b-bf04-da0b270d6cfb)

### Version

Master

### Device Info

OS: Android 15
Device: My favorite enslaved emulator

--- a/app/lib/applets/study_groups/student/parser.dart
+++ b/app/lib/applets/study_groups/student/parser.dart
@@ -76,29 +76,16 @@ class StudyGroupsStudentParser extends AppletParser<List<StudentStudyGroups>> {
 
     // Exams parse tbody
     List<List<String>> examData = [];
+    RegExp dateRegex = RegExp(r'.{2}, \d{2}\.\d{2}\.\d{4}');
     examTable.querySelectorAll('tbody tr').forEach((element) {
       List<String> examRow = [];
       // Check if first element contains a date
-      if (element
-          .querySelector('td')!
-          .text
-          .trim()
-          .contains(RegExp(r'\d{2}\.\d{2}\.\d{4}'))) {
-        element.querySelectorAll('td').forEach((element) {
-          // if there is a span with the class "next" it is the next exam
-          if (element.querySelector('span') != null) {
-            if (element.querySelector('span')!.classes.contains('next')) {
-              // Exclude the span from the text
-              String wholeText = element.text.trim();
-              String text = wholeText
-                  .substring(
-                      0,
-                      wholeText.length -
-                          element.querySelector('span')!.text.length)
-                  .trim()
-                  .replaceAll('\n', '');
-              examRow.add(text);
-            }
+      if (element.querySelector('td')!.text.trim().contains(dateRegex)) {
+        element.querySelectorAll('td').asMap().forEach((index, element) {
+          if (index == 0) {
+            RegExpMatch? match = dateRegex.firstMatch(element.text.trim());
+            // Match must be true, because of above if
+            examRow.add(match!.group(0)!);
           } else {
             examRow.add(element.text.trim());
           }

--- a/app/lib/applets/study_groups/student/parser.dart
+++ b/app/lib/applets/study_groups/student/parser.dart
@@ -85,7 +85,23 @@ class StudyGroupsStudentParser extends AppletParser<List<StudentStudyGroups>> {
           .trim()
           .contains(RegExp(r'\d{2}\.\d{2}\.\d{4}'))) {
         element.querySelectorAll('td').forEach((element) {
-          examRow.add(element.text.trim());
+          // if there is a span with the class "next" it is the next exam
+          if (element.querySelector('span') != null) {
+            if (element.querySelector('span')!.classes.contains('next')) {
+              // Exclude the span from the text
+              String wholeText = element.text.trim();
+              String text = wholeText
+                  .substring(
+                      0,
+                      wholeText.length -
+                          element.querySelector('span')!.text.length)
+                  .trim()
+                  .replaceAll('\n', '');
+              examRow.add(text);
+            }
+          } else {
+            examRow.add(element.text.trim());
+          }
         });
         examData.add(examRow);
       }

